src/clj/puppetlabs/services/jruby/jruby_puppet_core.clj

```diff
@@ -31,7 +32,19 @@
   classpath.
 
   See also:  http://jruby.org/apidocs/org/jruby/runtime/load/LoadService.html"
-  "puppetserver-lib")
+  (let [jruby-version Constants/VERSION]
+    (if (< (count jruby-version) 2)
+      (throw (Exception. (str "Unexpected jruby-version format: " jruby-version)))
+      ;; Under JRuby 9k (and presumably later JRuby versions as well), JRuby
+      ;; will fail to find resources from the classpath unless they are prefixed
+      ;; with "classpath:/".  JRuby 1.x will generally find resources when
+      ;; prefixed with "classpath:/" as well, but an invocation of JRuby's irb
+      ;; will error out with "java.lang.NullPointerException: 'in' is null"
+      ;; when the "classpath:/" prefix is included.  The code below ensures
+      ;; that the "classpath:/" prefix is only omitted for JRuby 1.x.
+      (str (if (not= "1." (subs jruby-version 0 2))
+             "classpath:/")
+           "puppetserver-lib"))))
```

Comment:
Doing a substring match makes me mildly wary, but we're pretty confident about what the version string will contain I guess